### DISCUSSION TITLE
how necessary is it to use ObjectId?

### DISCUSSION THREAD

**Discussion by jimmywarting** [Automattic/mongoose#11471] [Q&A] [answered]
I'm seeing stuff like this being used all over a project i started to help out in: 

```js
const ObjectId = require('mongoose').Types.ObjectId
const someModel = require('../userModel.js')

function handle (req, res) {
  await someModel.find({ _id: ObjectId(req.body.id) })
}
```

it's quite bloated. it seems perfectly fine to do: `  await someModel.find({ _id: req.body.id })` where `req.body.id` is just a string...

so my question is how often do you need to use ObjectId and when is it okey to just use a string?
i would assume only when you are creating your schema. that is when you should use it. other times it's okey to just use plain strings?

**Comment by vkarpov15** [comment, marked as answer]
As long as `_id` is an ObjectId in `userModel`'s schema, you shouldn't have to do `ObjectId()`. The below should be fine:

```javascript
await someModel.find({ _id: req.body.id });
```

By default, schemas have an `_id` of type ObjectId. So if there's no `_id` in `userModel`'s schema, the above code should still work.

Mongoose casts query filters to match your schema, so it can do things like convert strings to numbers or strings to ObjectIds: https://mongoosejs.com/docs/tutorials/query_casting.html
